### ISSUE TITLE
Q/feature req: how to only train selected days rows?

### ISSUE THREAD

**Issue by mw66** [Nixtla/neuralforecast#1411]
### Description

Suppose the input df has all the continous calendar days. And the look back window is 7 days (e.g Monday,  Tuesday,  ... Sunday), but we only want to train the Sunday rows (to predict only the next Monday value). 

Simple filtering only Sunday rows to create a new df as training data won't work: because after filtering,  the new df only has Sunday rows left, the other look back window days (Monday , ... Saturday) are gone!

How to handle such situation in neuralforecast?

Thanks.

**Comment by mw66** [comment]
Any update of this feature request?

**Comment by W057** [comment]
Hi! This functionality has been implemented in PR #1344 via the `training_data_availability_threshold` parameter. You can use it like:

```python
model = NHITS(
    h=12,
    input_size=24,
    training_data_availability_threshold=0.2,  # Require at least 20% data availability
    max_steps=100
)
```

- A single float (0.0 to 1.0) applies the same threshold to both insample and outsample windows
- A list/tuple of two floats `[insample_fraction, outsample_fraction]` allows separate thresholds

This filters out training windows that don't meet the minimum data availability requirement, which effectively allows training only on rows/periods with sufficient data.

Could this issue be closed?

**Comment by mw66** [comment]
Not really, in the OP example, how to specify: only train the Sunday rows?

**Comment by W057** [comment]
You're right, `training_data_availability_threshold` doesn't cover that case — it filters by data availability percentage, not by content.

For training only on specific days (e.g., Sundays), you can filter your DataFrame before passing it to `fit()`:

```python
df_sundays = df[df['ds'].dt.dayofweek == 6]
nf.fit(df=df_sundays)
```

The model will then only learn from those rows.

**Comment by mw66** [comment]
> You're right, `training_data_availability_threshold` doesn't cover that case — it filters by data availability percentage, not by content.
> 
> For training only on specific days (e.g., Sundays), you can filter your DataFrame before passing it to `fit()`:
> 
> df_sundays = df[df['ds'].dt.dayofweek == 6]
> nf.fit(df=df_sundays)
> The model will then only learn from those rows.

With such filtering, the training data X in the look back window will be gone: it is the previous 7 days (e.g Monday, Tuesday, ... Sunday).

What left is the past 7 Sundays as training data X, totally changed its meaning.

**Comment by W057** [comment]
That's a good point — filtering the DataFrame would break the lookback window structure, since the model needs the full sequence of previous days as input features.

What you're describing is essentially "target masking": keeping the full input window intact but only computing the training loss on specific time steps (e.g., Sundays). NeuralForecast doesn't currently support this natively — there's no parameter to specify which time steps should contribute to the loss during training.

This would be a valid feature request, something like a `sample_weight` or `target_mask` argument that lets you control which positions in the forecast window are used for loss computation.

**Comment by mw66** [comment]
I think we can add a parameter, e.g. `only_train_col` to allow user flag bool value in the df.

If not, specified, train every rows. Otherwise, train rows with that col is True.
